spec/integration/entangler_spec.rb

```diff
@@ -62,5 +62,133 @@
         count(:destination, 'common', 'inserted').must_equal(0)
       end
     end
+
+    describe 'entanglement with bombarding long running queries on specific tables' do
+      before(:each) do
+        Lhm::Entangler.const_set('TABLES_WITH_LONG_QUERIES_OLD', Lhm::Entangler::TABLES_WITH_LONG_QUERIES)
+        Lhm::Entangler.const_set('TABLES_WITH_LONG_QUERIES', 'origin')
+        execute("insert into origin (common) values ('inserted')")
+        Lhm::Entangler.const_set('LONG_QUERY_TIME_THRESHOLD_OLD', Lhm::Entangler::LONG_QUERY_TIME_THRESHOLD)
+        Lhm::Entangler.const_set('LONG_QUERY_TIME_THRESHOLD', 3)
+      end
+
+      after(:each) do
+        Lhm::Entangler.const_set('TABLES_WITH_LONG_QUERIES', Lhm::Entangler::TABLES_WITH_LONG_QUERIES_OLD)
+        Lhm::Entangler.const_set('TABLES_WITH_LONG_QUERIES_OLD', nil)
+        Lhm::Entangler.const_set('LONG_QUERY_TIME_THRESHOLD', Lhm::Entangler::LONG_QUERY_TIME_THRESHOLD)
+        Lhm::Entangler.const_set('LONG_QUERY_TIME_THRESHOLD_OLD', nil)
+      end
+
+      describe 'with long running query killing env var enabled' do
+        before do
+          ENV['LHM_KILL_LONG_RUNNING_QUERIES'] = 'true'
+        end
+
+        after do
+          ENV.delete('LHM_KILL_LONG_RUNNING_QUERIES')
+        end
+
+        it 'kills long running queries involving the origin table and does not block trigger related actions' do
+          begin
+            threads = []
+
+            puts "spawning threads to spam queries..."
+            query_spawning_thread = Thread.new do
+              loop do
+                sleep(Random.rand(1.5))
+                trd = Thread.new do
+                  connection = ActiveRecord::Base.connection
+                  connection.execute('select sleep(1000) from `origin`;')
+                end
+                threads << trd
+              end
+            end
+            threads << query_spawning_thread
+
+            puts "attempting trigger actions with bombarding queries..."
+            @entangler.run do
+              trigger_count = execute("select count(*) from information_schema.triggers where event_object_table = 'origin'").to_a.flatten.first
+              assert_equal trigger_count, 3
+            end
+
+            trigger_count = execute("select count(*) from information_schema.triggers where event_object_table = 'origin'").to_a.flatten.first
+            assert_equal trigger_count, 0
+          ensure
+            query_spawning_thread.terminate
+            puts "stopping query spamming..."
+            sleep(3)
+
+            puts "cleaning up rogue long queries..."
+            ActiveRecord::Base.connection.reconnect!
+            ids = ActiveRecord::Base.connection.execute("select id from information_schema.processlist where info like '\%sleep(1000)%' and time > 1").to_a.flatten
```

Comment:
Put `select sleep(1000) from `origin` in a variable and target the query specifically.